### ISSUE TITLE
Question: Why is WidenLiteral<T> intersected with {}?

### ISSUE THREAD

**Issue by petetnt** [total-typescript/ts-reset#37]
Sorry for the spam, but here's a question:

The Array and Set.has helpers have this kind of lines:

```
    searchElement: T | (TSReset.WidenLiteral<T> & {}),

// and

  has(value: T | (TSReset.WidenLiteral<T> & {})): boolean;
```

I assume the purpose is the [same as in this tweet (prettifying the output of TSReset.WidenLiteral<T>)](https://twitter.com/mattpocockuk/status/1622730173446557697) but I am not seeing any differences with or without it.

Are there scenarios where it would have some effect?

**Comment by mattpocock** [comment]
No! It's to enable autocomplete on members of the literal inside has.

So, you don't lose any autocomplete on Set.has, but it still allows the wider literal.
